### PR TITLE
Climate check ERA5 (1991-2020 baseline)

### PR DESCRIPTION
**Pull request by veronikasamborska1994**
hey @pabloarosado! 

I am making a few changes to the dataset. 

- Calculating global averages based on the latitude weighting and getting rid of our OWID based regional aggregates. 
- Using the same default period as suggested here -  https://climate.copernicus.eu/surface-air-temperature-january-2024 (the resulting values match which is a good sign). 
- I also want to regularly update the dataset (on a monthly basis) so I want to move everything to the "latest" folder eventually but the CDS mention that they are in the process of updating their database so for now I only added the snapshot "latest" directory with February data. 
- I am also updating some metadata which was a bit inaccurate as it turns out I've actually been using the [ERA5 data](https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-single-levels-monthly-means?tab=overview), not ERA5 land data. I've tried playing around with the ERA5 dataset but it has some issues (possibly because of their current updates) and we've decided with Hannah that for now we will use the ERA5 data.

### CODE REVIEW

**Review by pabloarosado — APPROVED** [review]
Looks good! I made some suggestions (in particular, against using "latest" for snapshots).

**Comment by pabloarosado** [comment]
Another small suggestion regarding `etl/steps/data/meadow/climate/2023-12-20/surface_temperature.py`. Instead of having a generic Exception, I think you could have the exact exception you are encountering, which seems to be `NoDataInBounds` (you'd need to add `from rioxarray.exceptions import NoDataInBounds` at the beginning). Otherwise, there may be other unknown issues that you are hiding without raising any error.
And regarding the 54 small countries that fail, we can look into some strategies to add data for them (but we can do that in a separate PR).

File: etl/steps/data/garden/climate/2023-12-20/surface_temperature.meta.yml
Comment (veronikasamborska1994) [reply etl/steps/data/garden/climate/2023-12-20/surface_temperature.meta.yml:21]: thanks! yep, just a typo

File: etl/steps/data/garden/climate/2023-12-20/surface_temperature.meta.yml
Comment (veronikasamborska1994) [reply etl/steps/data/garden/climate/2023-12-20/surface_temperature.meta.yml:20]: also forgot to fix this. thanks for spotting it!

File: etl/steps/data/garden/climate/2023-12-20/surface_temperature.py
Comment (veronikasamborska1994) [reply etl/steps/data/garden/climate/2023-12-20/surface_temperature.py:33]: don't think this is necessary. the variable name is not misleading and states exactly what this is is

File: etl/steps/data/grapher/climate/2023-12-20/surface_global_monthly_anomaly.py
Comment (veronikasamborska1994) [reply etl/steps/data/grapher/climate/2023-12-20/surface_global_monthly_anomaly.py:23]: yep, exactly. I will change this in the metadata now!

File: snapshots/climate/2023-12-20/surface_temperature.gz.dvc
Comment (veronikasamborska1994) [reply snapshots/climate/2023-12-20/surface_temperature.gz.dvc:3]: no, it is actually 1940  --> https://cds.climate.copernicus.eu/cdsapp#!/dataset/reanalysis-era5-single-levels-monthly-means?tab=overview

Comment (pabloarosado) [reply snapshots/climate/2023-12-20/surface_temperature.gz.dvc:3]: The charts and datasets we currently have in grapher start in 1950 (it's even written as the title of the datasets, e.g. [this one](https://owid.cloud/admin/datasets/6352)). If now they start in 1940, then you should change the metadata accordingly. Or are we removing the data for the first decade for some reason?

File: etl/steps/data/garden/climate/2023-12-20/surface_temperature.py
Comment (pabloarosado) [reply etl/steps/data/garden/climate/2023-12-20/surface_temperature.py:33]: I'm not saying it's misleading. But if you read "temperature_2m" you don't know straight away what the 2m mean. And given that there are no other temperatures, it's not strictly necessary. But again, it's not very important.

File: etl/steps/data/garden/climate/2023-12-20/surface_temperature.countries.json
Comment (pabloarosado) [reply etl/steps/data/garden/climate/2023-12-20/surface_temperature.countries.json:198]: I don't mind using "Global" within the scripts, if you think that avoids confusion. But in the resulting garden and grapher datasets, as well as in the charts, it would make more sense to use "World" because that's what we use everywhere. We use it for example in [the climate change explorer](https://ourworldindata.org/explorers/climate-change). It would be confusing to have sometimes "World" and sometimes "Global". I don't think we have stated anywhere that "World" means land (although for most datasets that's what it implies).

Comment (veronikasamborska1994) [reply etl/steps/data/garden/climate/2023-12-20/surface_temperature.countries.json]: hey @pabloarosado! thanks for taking a look at the pull request! I've now implemented the changes but need to re-run the snapshot data upload (because of the directory change) which is taking a while because of the Copernicus API queues during the day. It shouldn't change anything in the rest of the code though. Could you please take a look at the changes you've asked for and approve if you are happy with them so I can merge the pull request as soon I get the data downloaded (which will likely be in the evening when the API queues are shorter) without having to wait till tmrw.

Comment (veronikasamborska1994) [reply etl/steps/data/garden/climate/2023-12-20/surface_temperature.countries.json]: also, I think using "Global" is better for the reasons I've explained above but since it would be more effort for you to change it then for me I have renamed it to "World" now.

File: snapshots/climate/2023-12-20/surface_temperature.gz.dvc
Comment (veronikasamborska1994) [reply snapshots/climate/2023-12-20/surface_temperature.gz.dvc:3]: I was originally mistakenly only extracting data for 1950 and above but I've fixed it now. I've changed the metadata fields in the datasets accordingly as well.

File: etl/steps/data/garden/climate/2023-12-20/surface_temperature.countries.json
Comment (veronikasamborska1994) [reply etl/steps/data/garden/climate/2023-12-20/surface_temperature.countries.json]: p.s. the unit tests will be failing because I am still uploading the dataset to snapshot but I will check that everything is ok before merging.

File: snapshots/climate/2023-12-20/surface_temperature.gz.dvc
Comment (pabloarosado) [reply snapshots/climate/2023-12-20/surface_temperature.gz.dvc:3]: Good, thanks for fixing that. Now that you are changing it, I think in general it would be better to not include the coverage (in this case 1940) in the titles, since that's something that could eventually change.

Comment (veronikasamborska1994) [reply snapshots/climate/2023-12-20/surface_temperature.gz.dvc:3]: ok done

**Pull request merged**
